ecloud/data_source_active_directory.go

```diff
@@ -0,0 +1,68 @@
+package ecloud
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/ukfast/sdk-go/pkg/connection"
+	ecloudservice "github.com/ukfast/sdk-go/pkg/service/ecloud"
+)
+
+func dataSourceActiveDirectory() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceActiveDirectoryRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"ad_domain_id": &schema.Schema{
```

Comment:
We don't need to reference `ad_domain_id` here, as this is a duplicate of the datasource ID

Reply:
removed :)